### PR TITLE
feat: add python and curl calls export options

### PR DESCRIPTION
**Pull request by gtarpenning**
https://wandb.atlassian.net/browse/WB-20314

this pr: 
- adds python code snippet to use the calls displayed in the table 
- adds curl snippet
- snippet construction should account for filters and expanded columns, as well as the manual call selection (checkboxes)

TODO: 
- stylize the code blocks (punt)

Screenshot 2024-08-20 at 12 33 04 PM
Screenshot 2024-08-20 at 12 33 08 PM

Screenshot 2024-08-20 at 12 33 28 PM

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx [hunk 1]
```diff
@@ -68,6 +77,25 @@ export const ExportSelector = ({
     callQueryParams.gridSort
   );
 
+  // TODO(gst): tabulate correct list of ref columns
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:80]: We currently determine whether or not a ref is expandable on a value-by-value basis. Ie, there is no way to tell if a given column contains refs, without checking every single value. As implemented, this is just a very inclusive heuristic that allows all columns that are equal to or start with one of the json columns. I believe these are the only columns that contain expandable refs. The current sever implementation won't error if you pass a column to expand that has a non-ref value (heterogeneous data means that _some_ might be refs, others not), so this at least shouldn't error.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx [hunk 2]
```diff
@@ -268,6 +330,31 @@ const DownloadGrid: FC<{
...
+        <ClickableOutlinedCardWithIcon
+          iconName="code-alt"
+          disabled={false}
+          onClick={() => setCodeMode('curl')}>
+          Use CURL
+        </ClickableOutlinedCardWithIcon>
+      </div>
+      {codeMode && (
+        <div className="mt-8 flex max-w-full items-center">
+          <CodeEditor
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:350]: I don't know how to properly style this... Might have to break out the component and weave-ify it?

Comment (jamie-rasmussen) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:350]: How do you want to style it?

**Review by jamie-rasmussen — APPROVED** [review]

**Review by jamie-rasmussen — COMMENTED** [review]
Actually, let's hold off on landing this - doesn't seem like filtering works?

**Comment by gtarpenning** [comment]
> Actually, let's hold off on landing this - doesn't seem like filtering works?

Sorry, yes I am seeing that this is not quite compatible with the new filtering scheme you craeted. I opened this before those went in, I must have missed something important... Good catch. @jamie-rasmussen

**Comment by gtarpenning** [comment]
> > Actually, let's hold off on landing this - doesn't seem like filtering works?
> 
> Sorry, yes I am seeing that this is not quite compatible with the new filtering scheme you craeted. I opened this before those went in, I must have missed something important... Good catch. @jamie-rasmussen

Okay fixed, thanks!

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx [hunk 3]
```diff
@@ -268,6 +330,31 @@ const DownloadGrid: FC<{
           Export to JSON
         </ClickableOutlinedCardWithIcon>
       </div>
+      <div className="mt-8 flex items-center">
+        <ClickableOutlinedCardWithIcon
+          iconName="python-logo"
```
[the remark this answers is not in the document]
Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:338]: right... Any suggestions on how to deal with that? Check for night mode and handle night mode colors differently explicity?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx [hunk 4]
```diff
@@ -68,6 +77,25 @@ export const ExportSelector = ({
     callQueryParams.gridSort
   );
 
+  // TODO(gst): tabulate correct list of ref columns
+  const refColumnsToExpand = useMemo(
+    () =>
+      visibleColumns.filter(col => {
+        for (const maybeRefCol of [
```
[the remark this answers is not in the document]
Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx]: slick

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamie-rasmussen) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:325]: Maybe something like:
`className={classNames({'night-aware': iconName.includes('logo')})}` on the `<Icon>`

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx [hunk 5]
```diff
@@ -314,6 +359,41 @@ const DownloadGrid: FC<{
...
+          <div className="flex w-full justify-end">
+            <Icon name="copy" size="small" />
+          </div>
+        </ClickableOutlinedCardWithIcon>
+        <div className="ml-8" />
+        <ClickableOutlinedCardWithIcon
+          iconName="code-alt"
+          onClick={() => {
+            setCodeMode('curl');
+            navigator.clipboard.writeText(curlText);
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:379]: I'm having mixed feelings about autocopy to clipboard, did this come from design?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx [hunk 6]
```diff
@@ -376,3 +456,115 @@ function initiateDownloadFromBlob(blob: Blob, fileName: string) {
   document.body.removeChild(anchor);
   URL.revokeObjectURL(downloadUrl);
 }
+
+function useMakeCodeText(
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:460]: These aren't hooks, right? Shouldn't name `use*`

**Review by jamie-rasmussen — APPROVED** [review]

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx (hunk 5 above)
Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx:379]: Okay, I see your point. Lets leave this out and then circle back with design. @jamie-rasmussen

**Pull request merged**